src/main/java/appeng/core/sync/packets/PacketMEInventoryUpdate.java

```diff
@@ -61,162 +57,168 @@
 
 	// input.
 	@Nullable
-	private final List<IAEItemStack> list;
+	private List<IAEItemStack> list = new LinkedList<IAEItemStack>();
 	// output...
-	private final byte ref;
-
-	@Nullable
-	private final ByteBuf data;
-	@Nullable
-	private final GZIPOutputStream compressFrame;
+	private byte ref;
 
 	private int writtenBytes = 0;
 	private boolean empty = true;
 
 	// automatic.
-	public PacketMEInventoryUpdate( final ByteBuf stream ) throws IOException
+	public PacketMEInventoryUpdate()
 	{
-		this.data = null;
-		this.compressFrame = null;
-		this.list = new LinkedList<IAEItemStack>();
-		this.ref = stream.readByte();
+		this( (byte) 0 );
+	}
 
-		// int originalBytes = stream.readableBytes();
+	// api
+	public PacketMEInventoryUpdate( byte ref )
+	{
+		this.ref = ref;
+	}
+
+	public void appendItem( IAEItemStack is ) throws IOException, BufferOverflowException
+	{
+		// is.writeToPacket( this.data );
+		this.list.add( is.copy() );
+	}
+
+	public int getLength()
+	{
+		return this.writtenBytes;
+	}
+
+	public boolean isEmpty()
+	{
+		return this.empty;
+	}
+
+	@Override
+	public void fromBytes( final ByteBuf buf )
+	{
+		this.ref = buf.readByte();
 
-		GZIPInputStream gzReader = new GZIPInputStream( new InputStream()
+		try
 		{
-			@Override
-			public int read() throws IOException
+			final GZIPInputStream gzReader = new GZIPInputStream( new InputStream()
 			{
-				if( stream.readableBytes() <= 0 )
+				@Override
+				public int read() throws IOException
 				{
-					return -1;
+					if( buf.readableBytes() <= 0 )
+					{
+						return -1;
+					}
+
+					return buf.readByte() & STREAM_MASK;
 				}
+			} );
 
-				return stream.readByte() & STREAM_MASK;
-			}
-		} );
+			final ByteBuf uncompressed = Unpooled.buffer( buf.readableBytes() );
+			byte[] tmp = new byte[TEMP_BUFFER_SIZE];
 
-		ByteBuf uncompressed = Unpooled.buffer( stream.readableBytes() );
-		byte[] tmp = new byte[TEMP_BUFFER_SIZE];
-		while( gzReader.available() != 0 )
-		{
-			int bytes = gzReader.read( tmp );
-			if( bytes > 0 )
+			while( gzReader.available() != 0 )
 			{
-				uncompressed.writeBytes( tmp, 0, bytes );
+				int bytes = gzReader.read( tmp );
+				if( bytes > 0 )
+				{
+					uncompressed.writeBytes( tmp, 0, bytes );
+				}
 			}
-		}
-		gzReader.close();
 
-		// int uncompressedBytes = uncompressed.readableBytes();
-		// AELog.info( "Receiver: " + originalBytes + " -> " + uncompressedBytes );
+			gzReader.close();
 
-		while( uncompressed.readableBytes() > 0 )
+			while( uncompressed.readableBytes() > 0 )
+			{
+				this.list.add( AEItemStack.loadItemStackFromPacket( uncompressed ) );
+			}
+		}
+		catch( IOException e )
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)